components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -883,6 +907,18 @@ void RRTMGPRadiation::run_impl (const double dt) {
       // Get IR 10.5 micron band for COSP
       auto idx_105 = rrtmgp::get_wavelength_index_lw(10.5e-6);
 
+      // Compute cloud-top diagnostics following AeroCOM recommendation
+      auto T_mid_at_cldtop       = real1d("T_mid_at_cldtop", ncol);
+      auto p_mid_at_cldtop       = real1d("p_mid_at_cldtop", ncol);
+      auto cldfrac_ice_at_cldtop = real1d("cldfrac_ice_at_cldtop", ncol);
+      auto cldfrac_liq_at_cldtop = real1d("cldfrac_liq_at_cldtop", ncol);
+      auto cldfrac_tot_at_cldtop = real1d("cldfrac_tot_at_cldtop", ncol);
+      auto cdnc_at_cldtop        = real1d("cdnc_at_cldtop", ncol);
+      rrtmgp::compute_aerocom_cloudtop(
+          ncol, nlay, t_lay, p_lay, p_del, z_del, qc, qi, cldfrac_tot, nc,
```

Comment:
I feel like I'm constantly puzzling over where we should use "cldfrac_tot" vs "cldfrac_tot_for_analysis", but note that "cldfrac_tot" is very sensitive to the presence of even very small amounts of ice (I think the threshold for cldfrac_tot = 1 is something like qi > 1e-12). So, I think in using cldfrac_tot here, you're making sure that your cldfrac is consistent with radiation (probably a good thing), but I do wonder if this is going to make your diagnostics look a little funny in the presence of ice. Have you thought about this?